app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java

```diff
@@ -356,11 +360,37 @@ public void run() {
     }
 
     @OnClick(R.id.mediaDetailImageViewSpacer)
-    public void launchZoomActivity(View view) {
+    public void launchZoomActivity(final View view) {
+        final boolean permission = PermissionUtils.
+            hasPermission(getActivity(), Manifest.permission.READ_EXTERNAL_STORAGE);
+
+        if (permission) {
+            launchZoomActivityAfterPermissionCheck(view);
+        }
+        else {
+            PermissionUtils.checkPermissionsAndPerformAction(getActivity(),
+                Manifest.permission.READ_EXTERNAL_STORAGE,
+                () -> {
+                    launchZoomActivityAfterPermissionCheck(view);
+                },
+                R.string.storage_permission_title,
+                R.string.read_storage_permission_rationale
+            );
+        }
+    }
+
+    /**
+     * launch zoom acitivity after permission check
+     * @param view as ImageView
+     */
+    private void launchZoomActivityAfterPermissionCheck(final View view) {
         if (media.getImageUrl() != null) {
-            Context ctx = view.getContext();
+            final Context ctx = view.getContext();
+            final Intent zoomableIntent = new Intent(ctx, ZoomableActivity.class);
+            zoomableIntent.setData(Uri.parse(media.getImageUrl()));
+            zoomableIntent.putExtra("Origin", "MediaDetail");
```

Comment:
Could you please rename to "MediaDetails"?

Reply:
Done